docs/how-to/service-protection-setup.md

```diff
@@ -0,0 +1,74 @@
+# Installing Kuadrant Service Protection into an existing OCM Managed Cluster
+
+## Introduction
+This walkthrough will show you how to install and setup the Kuadrant Operator into an [OCM](https://open-cluster-management.io/) [Managed Cluster](https://open-cluster-management.io/concepts/managedcluster/).
+
+## Prerequisites
+* Access to an Open Cluster Management (>= v0.11.0) Managed Cluster, which has already been bootstrapped and registered with a hub cluster
+  * We have [a guide](./kuadrant-hub-install.md) which covers this in detail
+  * Also see:
+    * https://open-cluster-management.io/getting-started/quick-start/
+    * https://open-cluster-management.io/concepts/managedcluster/
+* OLM will need to be installed into the ManagedCluster where you want to run the Kuadrant Service Protection components
+  * See https://olm.operatorframework.io/docs/getting-started/
+* Kuadrant uses Istio as a Gateway API provider - this will need to be installed into the data plane clusters
+  * See https://istio.io/v1.16/blog/2022/getting-started-gtwapi/
```

Comment:
@eguzki wondering if we have a better resource for installing istio and also would like to confirm the version you are testing against.
@jasonmadigan In that blog it also reference 0.5.2 of GWAPI wondering if we just want to point them at basic install

```
curl -L https://istio.io/downloadIstio | sh -
$ cd istio-<version>
$ ./bin/istioctl install --set profile=minimal -y
```

seems to be the relevant command

Reply:
hmm ok so using this install method may not be right way currently. As I recall we need to use the `IstioOperator` resource

Reply:
also wondering if they already have istio installed is there a way they can point kuadrant to it and is there any specific config we need them to apply to an existing install

Reply:
looking at the istio-system namesapce I see there is an existing IstioOperator resource called ```installed-state``` so perhaps we can change find a way to change the defualt Kuadrant looks at. Looks to an env ```ISTIOOPERATOR_NAME ```

Reply:
1.19 is the latest release. Will see does it work with that

Reply:
Using currently [ISTIOVERSION = 1.17.2](https://github.com/Kuadrant/kuadrant-operator/blob/7dcbc3dabbe13d2092adeaf018b2aaccb0eee23a/make/istio.mk#L11)

We do not have doc in service protection to install istio. We link to upstream istio installation guide. Istio installation is out of scope of kuadrant (a.k.a. `batteries not included`) currently.

For our dev/testing environment, what we currently do is 
* download `istioctl` CLI tool for the specific version.
* run: `istioctl operator init` to install the Istio controller
* Apply [IstioOperator CR](https://github.com/Kuadrant/kuadrant-operator/blob/7dcbc3dabbe13d2092adeaf018b2aaccb0eee23a/config/dependencies/istio/istio-operator.yaml) with the following customizations:
  * Default ingress gateway enabled 
  * No egree gateway
  * No cni
  * istio namespace: `istio-system`

From their [Install operator guide](https://istio.io/latest/docs/setup/install/operator/), the operator is deprecated. ( on the other hand, from ServiceMesh side I read that there is WIP new operator for Istio to converge OSSM and istio :shrug: ) 
```
Use of the operator for new Istio installations is discouraged in favor of the [Istioctl](https://istio.io/latest/docs/setup/install/istioctl) and [Helm](https://istio.io/latest/docs/setup/install/helm) installation methods. While the operator will continue to be supported, new feature requests will not be prioritized.
```
Thus, we need to update the installation method for the dev/testing env

Reply:
yeah although `istioctl install --set profile=minimal -y` still creates it

Reply:
As of today, only used to register Authorino (to enable AuthPolicy), if the Istio installation has some custom values other than default values, yes, they need to be set as env vars
```go
func controlPlaneProviderName() string {                                         
    return env.GetString("ISTIOOPERATOR_NAME", "istiocontrolplane")              
}                                                                                
                                                                                 
func controlPlaneConfigMapName() string {                                        
    return env.GetString("ISTIOCONFIGMAP_NAME", "istio")                         
}                                                                                
                                                                                 
func controlPlaneProviderNamespace() string {                                    
    return env.GetString("ISTIOOPERATOR_NAMESPACE", "istio-system")              
}                                                                                

```

Reply:
PR to allow these to be customised https://github.com/Kuadrant/multicluster-gateway-controller/pull/494

Reply:
@jasonmadigan once the above merges. We can add a section about ```changing the default istiooperator name``` 

To do that you will now install the add-on as normal. Then in the cluster you want to modify the name / location, you will have to set 

```
k annotate managedclusteraddon kuadrant-addon "addon.open-cluster-management.io/values"='{"IstioOperator":"installed-state"}' -n <managed cluster>
```
this will propigate down and update the Kuadrant operator

Reply:
pushed some new words to a new section for bring-your-own-istio scenarios.